cpp/include/nvtext/generate_ngrams.hpp

```diff
@@ -56,5 +60,34 @@ std::unique_ptr<cudf::column> generate_ngrams(
   cudf::string_scalar const& separator = cudf::string_scalar{"_"},
   rmm::mr::device_memory_resource* mr  = rmm::mr::get_default_resource());
 
+/**
+ * @brief Generates ngrams of characters within each string.
+ *
+ * Each character of a string used to build ngrams.
+ * Ngrams are not created across strings.
+ *
+ * ```
+ * ["ab", "cde", "fgh"] would generate bigrams as ["ab", "cd", "de", "fg", "gh"]
+ * ```
+ *
+ * The size of the output column will be the total number of ngrams generated from
+ * the input strings column.
+ *
+ * All null row entries are ignored and the output contains all valid rows.
+ *
+ * @throw cudf::logic_error if `ngrams < 2`
+ * @throw cudf::logic_error if there are not enough characters to generate any ngrams
+ *
+ * @param strings Strings column to produce ngrams from.
+ * @param ngrams The ngram number to generate.
+ *               Default is 2 = bigram.
+ * @param mr Device memory resource used to allocate the returned column's device memory.
+ * @return New strings columns of tokens.
+ */
+std::unique_ptr<cudf::column> generate_character_ngrams(
+  cudf::strings_column_view const& strings,
+  cudf::size_type ngrams              = 2,
+  rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
```

Comment:
Where do the defaults come from? Are they necessary?

Reply:
> Where do the defaults come from? Are they necessary?

The `ngram=2` default is to support bigrams which is the most common type of ngram to generate.